pyfar/dsp/dsp.py

```diff
@@ -1800,12 +1800,243 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1,
             f"Domain is '{domain}', but has to be 'time', 'freq',"
             " or 'freq_raw'.")
     data[data == 0] = np.finfo(float).eps
+
     if return_prefix is True:
         return log_prefix * np.log10(np.abs(data) / log_reference), log_prefix
     else:
         return log_prefix * np.log10(np.abs(data) / log_reference)
 
 
+def soft_limit_spectrum(signal, limit, knee, frequency_range=None,
+                        direction='above', log_prefix=None):
+    """
+    Soft limiting the magniude spectrum.
+
+    Soft limiting gradually increases the gain reduction to avoid
+    discontinuities in the data that would appear in hard limiting. The
+    transition betwwen the magnitude where no limiting is applied to the
+    magnitude where the limiting reaches its full effect is termed `knee`
+    (see examples below).
+
+    Note that the limiting is applied on `signal.freq`, i.e. the data after
+    the FFT normalization.
+
+    Parameters
+    ----------
+    signal : Signal, FrequencyData
+        The input data
+    limit : number, Signal, FrequencyData
```

Comment:
would it make sence to add an example with limit as an object?

Reply:
Good idea. I did that. And as a consequence I decided to remove the possibility to pass signals. I noted that the limit should be passed in dB which does not make sense for signals. The other possibility would have been to convert `signal.freq` to dB, but I thought this would make the function behavior less clear.